FWCore/Framework/interface/EventSetup.h

```diff
@@ -94,28 +101,27 @@ namespace edm {
 
       /** can directly access data if data_default_record_trait<> is defined for this data type **/
       template< typename T>
-         void getData(T& iHolder) const {
-            typedef typename T::value_type data_type;
-            typedef typename eventsetup::data_default_record_trait<data_type>::type RecordT;
-            const RecordT& rec = this->get<RecordT>();
-            rec.get(iHolder);
+         bool getData(T& iHolder) const {
+            return getData("", iHolder);
```

Comment:
Won't this require an `std::string` to be created and destroyed each call?

Reply:
Yes.  It's a cost-benefit analysis of having a single point of maintenance for the `getData` overload set vs. the overhead of constructing an empty string.  I could make an empty-string `static const` data member (function) that would avoid the extra construction each time.  That's arguably more efficient, but the extra maintainability may not be worth it.